recipes/llvm/all/conanfile.py

```diff
@@ -0,0 +1,791 @@
+                        "Configured compiler.libcxx=libc++ will fail in test_package linking. If you want to try it set enable_unsafe_mode=True")
+            elif is_msvc(self):
+                # TODO probably migration needed: https://github.com/conan-io/tribe/blob/main/design/032-msvc_support.md
+                if Version(self.settings.compiler.version) < Version("16.4"):
+                    raise ConanInvalidConfiguration(
+                        "An up to date version of Microsoft Visual Studio 2019 or newer is required. If you want to try it set enable_unsafe_mode=True")
+
+    # XXX Still unsure if we should even check for this at all, errors like this would need a lot of fine tuning for each environment to be correct.
+    # import for Apt doesn't satisfy: E9011(conan-import-tools)
+    # def system_requirements(self):
+        # if self.is_linux():
+        #     # from conan.tools.system.package_manager import Apt
+        #     if self.options.get_safe('with_runtime_compiler-rt', False):
+        #         # apt should work for ubuntu/debian if apt is not installed this is skipped
+        #         if Apt(self).check(["libc6-dev-i386"]):
+        #             raise ConanInvalidConfiguration(
+        #                 "For compiler-rt you need the x86 header bits/libc-header-start.h, please install libc6-dev-i386")
+
+    def requirements(self):
+        if self.options.with_ffi:
+            # no version requirement in llvm 13-17
+            self.requires('libffi/[>3.4.0 <4.0.0]')
+        if self.options.get_safe('with_zlib', False):
+            # no version requirement in llvm 13-17
+            self.requires('zlib/[>1.2.0 <2.0.0]')
+        if self.options.get_safe('with_xml2', False):
+            # min version requirement from llvm 13-17
+            self.requires('libxml2/[>=2.5.3 <3.0.0]')
+        if self.options.get_safe('with_z3', False):
+            # min version requirement from llvm 13-17
+            self.requires('z3/[>=4.7.1 <5.0.0]')
+        if self.options.get_safe('with_curl', False):
+            # no version requirement in llvm 14-17
+            self.requires('libcurl/[>=7.76.0 <9.0.0]')
+        if self.options.get_safe('with_zstd', False):
+            # no version number in llvm 15-17
+            self.requires('zstd/[>=1.3.5 <2.0.0]')
+        if self.options.get_safe('with_httplib', False):
+            # no version number in llvm 15-17
+            self.requires('cpp-httplib/[>=0.5.4 <1.0.0]')
+
+    def build_requirements(self):
+        # Older cmake versions may have issues generating the graphviz output used
+        # to model the components
+        self.tool_requires("cmake/[>=3.21.3 <4.0.0]")
+        self.tool_requires("ninja/[>=1.10.0 <2.0.0]")
+
+    def generate(self):
+        tc = CMakeToolchain(self, "Ninja")
+        tc.generate()
+
+    def build(self):
+        cmake = self._cmake_configure()
+        cmake.build()
+
+    def _cmake_configure(self):
```

Comment:
This should be migrated to use the toolchain in the `generate` method instead of the CMake `configure` method.